application/src/main/kotlin/net/corda/v5/application/persistence/PersistenceService.kt

```diff
@@ -116,60 +115,43 @@ interface PersistenceService {
      *
      * @param queryName the name of the named query registered in the persistence context.
      * @param namedParameters the named parameters to be set in the named query.
-     * @param postFilter the filter to be applied after named query execution.
-     * @param R the type of the results
+     * @param R the type of the results.
+     * @param limit the maximum number of results to return.
      * @return Cursor configured to poll data for this named query.
      * @throws CordaPersistenceException if an error happens during query operation
      */
     @Suspendable
     fun <R> query(
         queryName: String,
         namedParameters: Map<String, Any>,
-        postFilter: NamedQueryFilter
+        limit: Int,
     ): Cursor<R>
 
     /**
-     * Execute a named query in a single transaction and apply post-processing to the results. Casts results to the specified type [R].
+     * Execute a named query in a single transaction, skip results to the specified [offset] and limit the number of returned results. Casts results to the specified type [R].
      *
      * @param queryName the name of the named query registered in the persistence context.
      * @param namedParameters the named parameters to be set in the named query.
-     * @param postProcessorName the name of the post-processor that will process named query results.
-     * @param R the type of the results
-     * @return Cursor configured to poll data for this named query.
-     * @throws CordaPersistenceException if an error happens during query operation
-     */
-    @Suspendable
-    fun <R> query(
-        queryName: String,
-        namedParameters: Map<String, Any>,
-        postProcessorName: String
-    ): Cursor<R>
-
-    /**
-     * Execute a named query in a single transaction and apply post-filtering and post-processing to the results. Casts results to the specified type [R].
-     *
-     * @param queryName the name of the named query registered in the persistence context.
-     * @param namedParameters the named parameters to be set in the named query.
-     * @param postFilter the filter to be applied after named query execution.
-     * @param postProcessorName the name of the post-processor that will process named query results.
-     * @param R the type of the results
+     * @param R the type of the results.
+     * @param offset the index of the first result in the query to return.
+     * @param limit the maximum number of results to return.
      * @return Cursor configured to poll data for this named query.
      * @throws CordaPersistenceException if an error happens during query operation
      */
     @Suspendable
     fun <R> query(
         queryName: String,
         namedParameters: Map<String, Any>,
-        postFilter: NamedQueryFilter,
-        postProcessorName: String
+        offset: Int,
+        limit: Int,
     ): Cursor<R>
```

Comment:
I didn't add one that has just offset, as I was thinking limit by itself may make sense, but maybe there aren't many use cases for using offset without limit.
Happy to add one though.

Reply:
Agreed, don't think it makes sense by itself,